shell/platform/darwin/macos/framework/Source/FlutterCompositor.mm

```diff
@@ -23,103 +19,62 @@
-      CFRelease(user_data);
-    }
-  };
-
-  return true;
-}
-
-bool FlutterCompositor::CollectBackingStore(const FlutterBackingStore* backing_store) {
-  // If we allocated this MTLTexture ourselves, user_data is not null, and we will need
-  // to release it manually.
-  if (backing_store->metal.texture.user_data != nullptr &&
-      backing_store->metal.texture.texture != nullptr) {
-    CFRelease(backing_store->metal.texture.texture);
-  }
+  backing_store_out->metal.struct_size = sizeof(FlutterMetalBackingStore);
+  backing_store_out->metal.texture = surface.asFlutterMetalTexture;
   return true;
 }
 
 bool FlutterCompositor::Present(uint64_t view_id,
                                 const FlutterLayer** layers,
                                 size_t layers_count) {
-  FlutterView* view = GetView(view_id);
+  FlutterView* view = [view_provider_ getView:view_id];
   if (!view) {
     return false;
   }
 
-  SetFrameStatus(FrameStatus::kPresenting);
-
-  bool has_flutter_content = false;
-  for (size_t i = 0; i < layers_count; ++i) {
-    const auto* layer = layers[i];
-    FlutterBackingStore* backing_store = const_cast<FlutterBackingStore*>(layer->backing_store);
-
-    switch (layer->type) {
-      case kFlutterLayerContentTypeBackingStore: {
-        if (backing_store->metal.texture.user_data) {
-          FlutterIOSurfaceHolder* io_surface_holder =
-              (__bridge FlutterIOSurfaceHolder*)backing_store->metal.texture.user_data;
-          IOSurfaceRef io_surface = [io_surface_holder ioSurface];
-          InsertCALayerForIOSurface(view, io_surface);
-        }
-        has_flutter_content = true;
-        break;
-      }
-      case kFlutterLayerContentTypePlatformView: {
-        PresentPlatformView(view, layer, i);
-        break;
+  NSMutableArray* surfaces = [NSMutableArray array];
+  for (size_t i = 0; i < layers_count; i++) {
+    FlutterLayer* layer = (FlutterLayer*)layers[i];
+    if (layer->type == kFlutterLayerContentTypeBackingStore) {
+      FlutterSurface* surface =
+          [view.surfaceManager lookupSurface:&layer->backing_store->metal.texture];
+      if (surface) {
+        FlutterSurfacePresentInfo* info = [[FlutterSurfacePresentInfo alloc] init];
+        info.surface = surface;
+        info.offset = CGPointMake(layer->offset.x, layer->offset.y);
+        info.zIndex = i;
+        [surfaces addObject:info];
       }
-    };
+    }
   }
 
-  return EndFrame(has_flutter_content);
+  [view.surfaceManager present:surfaces
+                        notify:^{
+                          for (size_t i = 0; i < layers_count; i++) {
+                            FlutterLayer* layer = (FlutterLayer*)layers[i];
+                            if (layer->type == kFlutterLayerContentTypePlatformView) {
+                              PresentPlatformView(view, layer, i);
+                            }
+                          }
+                          [platform_view_controller_ disposePlatformViews];
```

Comment:
I searched and found nobody calling this method at all right now. Is it because the presenting is not properly implemented anyway?

Reply:
Do you mean calling `disposePlatformViews`? I'm not sure what you mean by nobody, `FlutterPlatformViewController` is only used on macOS.

On iOS for example the vies are disposed [here](https://github.com/flutter/engine/blob/7b4291548841f91dd2796803fc27e707f84ebf22/shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm#L634).

Reply:
@dkwingsmt it's because platform views aren't yet supported on macOS, so the lack of the dispose call comes down to the current state of platform views not being finished and/or an oversight.